pulp_rpm/app/modulemd.py

```diff
@@ -0,0 +1,87 @@
+import json
+import tempfile
+
+from django.db.utils import IntegrityError
+from pulpcore.app.models.content import Artifact
+from pulp_rpm.app.constants import PULP_MODULE_ATTR
+
+import gi
+gi.require_version('Modulemd', '2.0')
+from gi.repository import Modulemd as mmdlib  # noqa: E402
+
+
+def _create_snippet(snippet_string):
+    """Create snippet of modulemd[-defaults] as artifact."""
+    tmp_file = tempfile.NamedTemporaryFile()
+    with open(tmp_file.name, 'w') as f:
+        f.write(snippet_string)
+    artifact = Artifact.init_and_validate(tmp_file.name)
+    try:
+        artifact.save()
+        return artifact
+    except IntegrityError:
+        artifact = Artifact.objects.get(sha256=artifact.sha256)
+        return artifact
+
+
+def _get_modulemd_names(modulemd_yaml, module_index):
+    """Get modulemd names."""
+    ret, fails = module_index.update_from_string(modulemd_yaml, True)
```

Comment:
As firstly used for upload which can't pass file stream directly (as using pulptemporaryupload class) string used instead

Reply:
Right. I think since upload will be partially handled by the core, it's very likely that by the time the plugin is called, you'll have a file you can refer to, so there is no need to read from it and pass a string to a library when you can just provide a file descriptor.